pkg/lib/synthesize/generate_test.go

```diff
@@ -0,0 +1,95 @@
+package synthesize
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+)
+
+func TestGenerateCSV(t *testing.T) {
+	csvlessBundlePath := "./testdata/csvless_bundle/prometheus_v0.22.2"
+	csvPrometheusBundleFilePath := "./testdata/csvless_bundle/prometheus_v0.22.2/manifests/olm.clusterserviceversion.yaml"
+	annotationsPrometheusBundleFilePath := "./testdata/csvless_bundle/prometheus_v0.22.2/metadata/annotations.yaml"
+	expectedCsvPrometheusBundleFilePath := "./testdata/csvless_bundle/prometheus_v0.22.2.clusterserviceversion.yaml"
+
+	utilrand.Seed(100)
+
+	// Test generating a CSV and annotations mediatype updates.
+	err := GenerateCSV(csvlessBundlePath)
+	require.NoError(t, err)
+	assertMediaType(t, annotationsPrometheusBundleFilePath, bundle.RegistryV1Type)
+	assert.Equal(t, loadCSV(t, expectedCsvPrometheusBundleFilePath), loadCSV(t, csvPrometheusBundleFilePath))
+
+	// Statically validate bundle with generated CSV.
+	logger := log.NewEntry(log.New())
+	bundleValidator := bundle.NewBundleValidator(logger)
+	err = bundleValidator.ValidateBundleFormat(csvlessBundlePath)
+	assert.NoErrorf(t, err, "error validating bundle format")
+	err = bundleValidator.ValidateBundleContent(csvlessBundlePath + "/manifests")
+	assert.NoErrorf(t, err, "error validating bundle content")
+
+	// reset test directory
+	copyFile(t, "./testdata/csvless_bundle/annotations.yaml", annotationsPrometheusBundleFilePath)
+	os.Remove(csvPrometheusBundleFilePath)
```

Comment:
It's probably best to avoid altering files in the `testdata` directory if we can help it.

Reply:
In this case, the copy is to ensure we have the input anootaion.yaml file and ensure thatr we changed its media type.

I can delete the remove here, but the remove in the second test case has to be there because we want to ensure that we rewrote the csv file.

Reply:
The intent of my comment was to point out that we should keep our test input separate from its output, both to keep things simple and to avoid haywire tests messing up the `testdata`.

We can address in a follow up.